src/server/node_func.c

```diff
@@ -1948,8 +1968,8 @@ fix_indirectness(resource *presc, struct pbsnode *pnode, int doit)
 
 			/* new is not indirect, was the original? */
 
-			if (porignalrc) {
-				if (porignalrc->rs_value.at_flags & ATR_VFLAG_INDIRECT) {
+			if (presc_assn) {
```

Comment:
Please add a comment that this is a hack that we are using resource-assigned to identify that the resource was an indirect resource because attribute's set function has already changed resources-available